core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/source/mysql/MysqlSource.java

```diff
@@ -0,0 +1,246 @@
+                    }
+                }
+                else{*/
+                if(predicate.getLimit()!=Integer.MAX_VALUE){
+                    ps.setObject(1, predicate.getLimit(), Types.INTEGER);
+                }
+                this.rs = ps.executeQuery();
+                start = false;}
+            //}
+			while (rs.next()) {
+			    List<IField> tb = new ArrayList();
+			    for(Attribute a: this.outputSchema.getAttributes()){
+			        if (a.getType() == AttributeType.STRING){
+			            String value = rs.getString(a.getName());
+			            value = value ==null? "":value;
+			            tb.add(new StringField(value));
+                    }else if (a.getType() == AttributeType.INTEGER){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0":value;
+                        tb.add(new IntegerField(Integer.valueOf(value)));
+                    }else if (a.getType() == AttributeType.DOUBLE){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0.0":value;
+                        tb.add(new DoubleField(Double.valueOf(value)));
+                    }else if (a.getType() == AttributeType.INTEGER){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0":value;
+                        tb.add(new IntegerField(Integer.valueOf(value)));
+                    }else if (a.getType() == AttributeType.DATE){
+                        Date value = rs.getDate(a.getName());
+                        if (value == null) {
+                            SimpleDateFormat fmt = new SimpleDateFormat("yyyy-MM-dd");
+                            try{
+                                value = fmt.parse("0000-00-00");
+                            }catch (ParseException e){
+                                System.out.println("parse error");
+                            }
+                        }
+                        tb.add(new DateField(value));
+                    }else if (a.getType() == AttributeType.DATETIME){
+                        Timestamp value = rs.getTimestamp(a.getName());
+
+                        if (value == null) {
+                            value = Timestamp.valueOf("0000-00-00 00:00:00.0");
+                        }
+                        tb.add(new DateTimeField(value.toLocalDateTime()));
+                    }
+                }
+                IField[] iFieldArray = tb.toArray(new IField[tb.size()]);
+                Tuple tuple = new Tuple(this.outputSchema, iFieldArray);
+			    cursor ++;
+			    
+			    return tuple;
+			}
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+        return null;
+    }
+    
+    public static String generateSqlQuery(MysqlSourcePredicate predicate) {
+        String query =  "\n" +
```

Comment:
Use paramters instead of concat strings,  see this tutorial and you'll understand: http://tutorials.jenkov.com/jdbc/preparedstatement.html